step-08/src/main/java/dev/langchain4j/quarkus/workshop/PromptInjectionGuard.java

```diff
@@ -4,15 +4,13 @@
 import io.quarkiverse.langchain4j.guardrails.InputGuardrail;
 import io.quarkiverse.langchain4j.guardrails.InputGuardrailResult;
 import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
 
 @ApplicationScoped
 public class PromptInjectionGuard implements InputGuardrail {
 
-    private final PromptInjectionDetectionService service;
-
-    public PromptInjectionGuard(PromptInjectionDetectionService service) {
-        this.service = service;
-    }
+    @Inject
+    PromptInjectionDetectionService service;
```

Comment:
Hm, actually this should not make any difference because [simplified constructor injection](https://quarkus.io/guides/cdi-reference#simplified-constructor-injection) is still a regular injection :thinking:

Reply:
I guess you're right and what does make the difference is having added the `@ApplicationScoped` annotation on the `PromptInjectionDetectionService`, correct?